lib/LANraragi/Model/Plugins.pm

```diff
@@ -243,8 +248,11 @@ sub exec_metadata_plugin {
             oneshot_param  => $oneshotarg
         );
 
-        my %newmetadata = $plugin->get_tags( \%infohash, @args );
+        my %newmetadata;
 
+        %newmetadata = $plugin->get_tags( \%infohash, @args );
+
+        # TODO: remove this block after changing all the metadata plugins
         #Error checking
         if ( exists $newmetadata{error} ) {
```

Comment:
Recently I saw a repository with some other plugins, so I'm unsure whether to remove this block of code without notice.
We should leave it for some time and add a warning in the documentation.

Reply:
Right might as well keep it in - It's not a lot of code anyway so it's better to avoid the breaking change.